src/main/kotlin/com/team4099/robot2023/subsystems/elevator/ElevatorIONEO.kt

```diff
@@ -0,0 +1,188 @@
+package com.team4099.robot2023.subsystems.elevator
+
+import com.revrobotics.CANSparkMax
+import com.revrobotics.CANSparkMaxLowLevel
+import com.revrobotics.SparkMaxPIDController
+import com.team4099.lib.math.clamp
+import com.team4099.robot2023.config.constants.Constants
+import com.team4099.robot2023.config.constants.ElevatorConstants
+import com.team4099.robot2023.subsystems.falconspin.MotorChecker
+import com.team4099.robot2023.subsystems.falconspin.MotorCollection
+import com.team4099.robot2023.subsystems.falconspin.Neo
+import org.littletonrobotics.junction.Logger
+import org.team4099.lib.units.base.Length
+import org.team4099.lib.units.base.Meter
+import org.team4099.lib.units.base.amps
+import org.team4099.lib.units.base.celsius
+import org.team4099.lib.units.base.inAmperes
+import org.team4099.lib.units.base.inPercentOutputPerSecond
+import org.team4099.lib.units.derived.DerivativeGain
+import org.team4099.lib.units.derived.ElectricalPotential
+import org.team4099.lib.units.derived.IntegralGain
+import org.team4099.lib.units.derived.ProportionalGain
+import org.team4099.lib.units.derived.Volt
+import org.team4099.lib.units.derived.inVolts
+import org.team4099.lib.units.derived.volts
+import org.team4099.lib.units.sparkMaxLinearMechanismSensor
+import kotlin.math.absoluteValue
+
+object ElevatorIONEO : ElevatorIO {
+
+    private val leaderSparkMax =
+        CANSparkMax(Constants.Elevator.LEADER_MOTOR_ID, CANSparkMaxLowLevel.MotorType.kBrushless)
+
+    private val leaderSensor =
+        sparkMaxLinearMechanismSensor(
+            leaderSparkMax,
+            ElevatorConstants.GEAR_RATIO,
+            ElevatorConstants.SPOOL_DIAMETER,
+            ElevatorConstants.VOLTAGE_COMPENSATION
+        )
+
+    private val followerSparkMax =
+        CANSparkMax(Constants.Elevator.FOLLOWER_MOTOR_ID, CANSparkMaxLowLevel.MotorType.kBrushless)
+
+    private val leaderPIDController: SparkMaxPIDController = leaderSparkMax.pidController
+
+    init {
+
+        // reseting motor
+        leaderSparkMax.restoreFactoryDefaults()
+        followerSparkMax.restoreFactoryDefaults()
+
+        leaderSparkMax.clearFaults()
+        followerSparkMax.clearFaults()
+
+        // basic settings
+        leaderSparkMax.enableVoltageCompensation(ElevatorConstants.VOLTAGE_COMPENSATION.inVolts)
+        followerSparkMax.enableVoltageCompensation(ElevatorConstants.VOLTAGE_COMPENSATION.inVolts)
+
+        leaderSparkMax.inverted = true
+
+        leaderSparkMax.setSmartCurrentLimit(ElevatorConstants.LEADER_STATOR_CURRENT_LIMIT.inAmperes.toInt())
+        followerSparkMax.setSmartCurrentLimit(ElevatorConstants.FOLLOWER_STATOR_CURRENT_LIMIT.inAmperes.toInt())
+
+        leaderSparkMax.openLoopRampRate = ElevatorConstants.RAMP_RATE.inPercentOutputPerSecond
```

Comment:
i think we should omit using ramp rate stuff. the effect it'll have on following motion profiles is unknown to me and i would prefer we just avoided having to consider it. we also want this to be fast so unncessary